internal/controllers/machine/machine_controller_phases.go

```diff
@@ -175,7 +175,7 @@ func (r *Reconciler) reconcileBootstrap(ctx context.Context, cluster *clusterv1.
 	log := ctrl.LoggerFrom(ctx, "cluster", cluster.Name)
 
 	// If the bootstrap data is populated, set ready and return.
-	if m.Spec.Bootstrap.DataSecretName != nil {
+	if m.Spec.Bootstrap.DataSecretName != nil && m.Spec.Bootstrap.ConfigRef == nil {
```

Comment:
Need this change so that the cert expiry continue working with backup and restore.

The cert expiry information in machine status needs to be updated after the machine object is restored.

Reply:
I'm not sure about this change. data secret can be generated by a boostrap config object (m.Spec.Bootstrap.ConfigRef != nil), and also in this case we want the condition to be set